src/diem/testing/miniwallet/app/offchain_api_v2.py

```diff
@@ -102,6 +116,46 @@ def _handle_offchain_payment_command(self, sender_address: str, request: Command
             account_id = self.app.store.find(Subaddress, subaddress_hex=subaddress).account_id
             self._create_payment_command(account_id, new_offchain_cmd, validate=True)
 
+    def _handle_offchain_reference_i_d_command(
+        self, request_sender_address: str, request: offchain.CommandRequestObject
+    ) -> Dict[str, Any]:
+        if request.command_type != CommandType.ReferenceIDCommand:
+            raise protocol_error(
+                ErrorCode.unknown_command_type,
+                f"unknown command_type: {request.command_type}",
+                field="command_type",
+            )
+        ref_id_command_object = from_dict(request.command, ReferenceIDCommandObject)
+
```

Comment:
it's better to create `validate_uniq_reference_id` method in app.py to validate, for example:

```def validate_uniq_reference_id(.....) -> None:```

so there becomes:
```
self.app.validate_uniq_reference_id(...)
self.app.validate_diem_id(...)
```
the code is more readable, and you don't need a comment to explain it.

`validate_diem_id` probably should be in the app.py, because when you send payment to internal account, will need it to validate the txn payee when creating the transaction.